### PR TITLE
Updated connectors to use context cache.

### PR DESCRIPTION
**Pull request by AndyButland**
Depends on [this PR](https://github.com/umbraco/Umbraco-Deploy/pull/673) in Deploy.

### CODE REVIEW

File: src/Umbraco.Deploy.Contrib.Connectors/Umbraco.Deploy.Contrib.Connectors.csproj [hunk 1]
```diff
@@ -5,10 +5,7 @@
         <RootNamespace>Umbraco.Deploy.Contrib.Connectors</RootNamespace>
         <TargetFramework>net472</TargetFramework>
     </PropertyGroup>
-
     <ItemGroup>
-        <PackageReference Include="UmbracoDeploy.Core">
-            <Version>4.1.2</Version>
-        </PackageReference>
+      <ProjectReference Include="..\..\..\Deploy-V4\src\Umbraco.Deploy\Umbraco.Deploy.csproj" />
```
Comment (ronaldbarendse) [inline src/Umbraco.Deploy.Contrib.Connectors/Umbraco.Deploy.Contrib.Connectors.csproj:9]: This should be updated to the package reference again when the new minor is released!

File: src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/BlockEditorValueConnector.cs [hunk 2]
```diff
@@ -8,20 +8,27 @@
...
         private readonly IContentTypeService _contentTypeService;
         private readonly Lazy<ValueConnectorCollection> _valueConnectorsLazy;
         private readonly ILogger _logger;
 
-        public virtual IEnumerable<string> PropertyEditorAliases => new[] { "Umbraco.BlockEditor" };
+        /// <inheritdoc />
+        public override IEnumerable<string> PropertyEditorAliases { get; } = new[]
+        {
+            "Umbraco.BlockEditor"
+        };
```
Comment (ronaldbarendse) [inline src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/BlockEditorValueConnector.cs:31]: There's no generic block editor (`Umbraco.BlockEditor` does not exist, only `Umbraco.BlockList` and soon `Umbraco.BlockGrid`) and this is an abstract class, so this should actually not implement/override the abstract property from the `ValueConnectorBase`...

Although removing this override is a breaking change, so is changing the base class to `ValueConnectorBase`, as the `ToArtifact()` and `FromArtifact()` methods without `IContextCache` are sealed and this previously declared them as virtual.

File: src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/MultiUrlPickerValueConnector.cs [hunk 3]
```diff
@@ -20,6 +22,13 @@ public class MultiUrlPickerValueConnector : IValueConnector
         // Used to fetch the udi from a umb://-based url
         private static readonly Regex MediaUdiSrcRegex = new Regex(@"(?<udi>umb://media/[A-z0-9]+)", RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
 
+        /// <inheritdoc />
+        public override IEnumerable<string> PropertyEditorAliases { get; } = new[]
+        {
+            "RJP.MultiUrlPicker",
```
Comment (ronaldbarendse) [inline src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/MultiUrlPickerValueConnector.cs:28]: This picker is only supported in Umbraco 7, so it should be safe to remove:
```suggestion
```

File: src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/NestedContentValueConnector.cs [hunk 4]
```diff
@@ -8,22 +8,32 @@
...
-        // Umbraco.NestedContent is Core NestedContent (introduced in v7.7)
-        public virtual IEnumerable<string> PropertyEditorAliases => new[] { "Our.Umbraco.NestedContent", "Umbraco.NestedContent" };
+        /// <inheritdoc />
+        /// <remarks>
+        /// Our.Umbraco.NestedContent is the original NestedContent package
+        ///  Umbraco.NestedContent is Core NestedContent (introduced in v7.7)
+        /// </remarks>
+        public override IEnumerable<string> PropertyEditorAliases { get; } = new[]
+        {
+            "Our.Umbraco.NestedContent",
```
Comment (ronaldbarendse) [inline src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/NestedContentValueConnector.cs:34]: This editor is also only supported in Umbraco 7, so it should be safe to remove:

```suggestion
```

File: src/Umbraco.Deploy.Contrib.Connectors/Umbraco.Deploy.Contrib.Connectors.csproj (hunk 1 above)
Comment (AndyButland) [reply src/Umbraco.Deploy.Contrib.Connectors/Umbraco.Deploy.Contrib.Connectors.csproj:9]: Oops, yes, will revert.  We'll need to bump the dependency too.

File: src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/BlockEditorValueConnector.cs (hunk 2 above)
Comment (ronaldbarendse) [reply src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/BlockEditorValueConnector.cs:31]: This is now fixed in https://github.com/umbraco/Umbraco.Deploy.Contrib/pull/59/commits/928afaf0d6d009aca01a1cfb19577de57a1be8ff 👍🏻

**Pull request merged**